lib/src/test/java/io/ably/lib/transport/HostsTest.java

```diff
@@ -83,4 +85,58 @@ public void hosts_fallback_custom_hosts() {
 
         assertThat(customHosts, IsIterableContainingInAnyOrder.containsInAnyOrder(resultList.toArray()));
     }
+
+    /**
+     * Expect that returned host is contained within custom host list in options
+     */
+    @Test
+    public void hosts_fallback_custom_hosts_options(){
+        List<String> customHosts    = Arrays.asList("F.ably-realtime.com", "G.ably-realtime.com", "H.ably-realtime.com", "I.ably-realtime.com", "J.ably-realtime.com", "K.ably-realtime.com");
+        ClientOptions options       = new ClientOptions();
+
+        options.setFallbackHosts(customHosts);
+        String host = Hosts.getFallback(null, options.getFallbackHosts());
+
+        assertTrue(options.getFallbackHosts().contains(host));
+    }
+
+    /**
+     * Expect that returned host is contained within default host list in options
+     */
+    @Test
+    public void hosts_fallback_no_custom_hosts_options(){
+        ClientOptions options = new ClientOptions();
+
+        String host = Hosts.getFallback(null, options.getFallbackHosts());
+
+        assertTrue(Defaults.HOST_FALLBACKS.contains(host));
+    }
+
+    /**
+     * Expect that returned host is null when setting an empty list in options
+     */
+    @Test
+    public void hosts_fallback_empty_custom_hosts_options(){
+        ClientOptions options = new ClientOptions();
+        options.setFallbackHosts(new ArrayList<>());
+
+        String host = Hosts.getFallback(null, options.getFallbackHosts());
+
+        assertThat(host, is(equalTo(null)));
+    }
+
+    /**
+     * Expect that returned host is null when setting null as fallback host list
+     */
+    @Test
+    public void hosts_fallback_null_custom_hosts_options(){
+        ClientOptions options = new ClientOptions();
+        options.setFallbackHosts(null);
```

Comment:
Sorry, but seems we are not communicating clearly, perhaps my fault.

Adding a setter / getter is not the solution given the IDL explicitly defines the attribute as `fallbackHosts`, see http://docs.ably.io/client-lib-development-guide/features/#TO3k6.  Unless there is a very good reason to deviate from the API that all client libraries use, we do not do this.  Also, adding a getter & setter is unnecessary given you could simply shuffle the `fallbackHosts` after the Client is initialised.  Additionally unfortunately with these changes the shuffling does not apply to the default fallback hosts now, see https://github.com/ably/ably-java/pull/141/commits/76d931f545c6cc588f67b42d0554c0e9a2b13677#diff-f111f958256bd5375e3c22f2b1fed74bR142.  Finally, I still do not think this test shows an end-to-end type test that we expect.  For example, how do I know that the client library actually uses those fallback hosts.  You are still testing the use of `Host.getFallback` to check if a fallback host is used, yet that is an internal implementation API.  So there is no guarantee that the fallback host is ever used unfortunately when the client library is actually unable to use the primary host.